envoy/common/execution_context.h

```diff
@@ -19,6 +28,13 @@ class ExecutionContext : NonCopyable {
   ExecutionContext() = default;
   virtual ~ExecutionContext() = default;
 
+  // Called when enters a scope in which |span| is active.
+  // Returns an object that can do some cleanup when exits the scope.
+  virtual Envoy::Cleanup onScopeEnter(Envoy::Tracing::Span& span) PURE;
+  // Called when enters a scope in which |filter_context| is active.
+  // Returns an object that can do some cleanup when exits the scope.
+  virtual Envoy::Cleanup onScopeEnter(const Http::FilterContext& filter_context) PURE;
```

Comment:
I am not sure which info that you want to extract from the active span or filter context? If you only want to ensure which filter is running or which span is active, then, maybe a pair of enum and string would be better. For example, you can use a enum to indicate the type of event, another string to indicate specific event like the filter name.
Then you can get a more general `onScopeEnter`.

Reply:
For the `Envoy::Tracing::Span` method, I'm doing something like below in my server:

```
Envoy::Cleanup MyEnvoyExecutionContext::onScopeEnter(Envoy::Tracing::Span& span) {
  auto my_span = dynamic_cast<MyEnvoyTracingSpan*>(&span);
  if (my_span == nullptr) {
    return Envoy::Cleanup::Noop();
  }
  // Use *my_span here
}
```

A string will not work in this case.